py/picca/pk1d/postproc_pk1d.py

```diff
@@ -862,12 +818,370 @@ def compute_average_pk_redshift(
     )
 
 
+def compute_and_fill_covariance(
+    compute_covariance,
+    compute_bootstrap,
+    weight_method,
+    mean_p1d_table,
+    nbins_k,
+    nbins_z,
+    p1d_groups,
+    number_worker,
+    number_bootstrap,
+    cov_table,
+):
+    """Compute the covariance and bootstrap covariance and fill the corresponding
+    cov_table variable
+
+    compute_covariance: Bool
+    If True, compute statistical covariance matrix of the mean P1D.
+    Requires  p1d_table to contain 'sub_forest_id', since correlations are computed
+    within individual forests.
+
+    compute_bootstrap: Bool
+    If True, compute statistical covariance using a simple bootstrap method.
+
+    weight_method: str,
+    Method to weight the data.
+
+    mean_p1d_table (array-like):
+    Table of mean 1D power spectra, with column 'meanPk'.
+
+    nbins_k (int):
+    Number of k bins.
+
+    nbins_z (int):
+    Number of z bins.
+
+    p1d_groups (array-like):
+    Individual p1d pixels grouped in the same wavenumber binning for all subforest
+
+    number_worker: int
+    Calculations of mean P1Ds and covariances are run parallel over redshift bins.
+
+    number_bootstrap: int
+    Number of bootstrap samples used if compute_bootstrap is True.
+
+    cov_table (array-like):
+    Covariance table to fill.
+
+    Return
+    ------
+    None
+    """
+
+
+    if compute_covariance:
+        userprint("Computation of covariance matrix")
+
+        func = partial(
+            compute_cov,
+            weight_method,
+            mean_p1d_table,
+            nbins_k,
+        )
+        if number_worker == 1:
+            output_cov = [func(*p1d_group) for p1d_group in p1d_groups]
+        else:
+            with Pool(number_worker) as pool:
+                output_cov = pool.starmap(func, p1d_groups)
+
+        for izbin in range(nbins_z):
+            covariance_array = output_cov[izbin]
+            i_min = izbin * nbins_k * nbins_k
+            i_max = (izbin + 1) * nbins_k * nbins_k
+            cov_table["covariance"][i_min:i_max] = covariance_array
+
+    if compute_bootstrap:
```

Comment:
is the bootstrap method tested? If yes, would be nice to know it's results, if no please at least add comments that this isn't tested yet...

Reply:
The bootstrap method works and gives very similar results than the classical covariance, which makes sense considering the number of chunks.
All the Y1 plot I am showing is with bootstrap